client/src/core/model-registry.js

```diff
@@ -392,6 +474,21 @@ class ModelRegistry {
         boundingBox.material = scaledBoundingBoxMaterial;
       }
 
+      if (shear) {
+        // We don't change the bounding box, matching AW behaviour
+        const shearFactor = shear.axes;
+        obj3d.geometry = this.applyShear(
+            obj3d.geometry,
+            shearFactor.x1,
+            shearFactor.y1,
+            shearFactor.z1,
+            shearFactor.x2,
+            shearFactor.y2,
+            shearFactor.z2,
+        );
```

Comment:
What about the `makeShear` method provided by three.js on `Matrix4` objects? it looks like it has all the parameters you need:
https://threejs.org/docs/#api/en/math/Matrix4.makeShear

Also, having to update vertices one by one like `applyShear` currently does is way more expensive than just modifying the transform matrix of the object.

```javascript
const shearMatrix = new Matrix4();
shearMatrix.makeShear(shearFactor.x1, ...); // maybe you need to pass them in a different order, check the doc
geometry.applyMatrix4(shearMatrix);
```